torch/testing/_internal/common_nn.py

```diff
@@ -871,6 +871,36 @@ def smoothl1loss_no_reduce_scalar_test():
         pickle=False)
 
 
+def smoothl1loss_beta_test():
+    t = torch.randn(2, 3, 4)
+    return dict(
+        fullname='SmoothL1Loss_beta',
+        constructor=wrap_functional(
+            lambda i: F.smooth_l1_loss(i, t.type_as(i), reduction='none', beta=0.5)),
+        cpp_function_call='''F::smooth_l1_loss(
+            i, t.to(i.options()), F::SmoothL1LossFuncOptions().reduction(torch::kNone), 0.5)''',
+        input_fn=lambda: torch.randn(2, 3, 4),
+        cpp_var_map={'i': '_get_input()', 't': t},
+        reference_fn=lambda i, *_:
+            loss_reference_fns['SmoothL1Loss'](i, t.type_as(i), reduction='none', beta=0.5),
+        pickle=False)
+
+
+def smoothl1loss_zero_beta_test():
+    t = torch.randn(2, 3, 4)
+    return dict(
+        fullname='SmoothL1Loss_beta',
+        constructor=wrap_functional(
+            lambda i: F.smooth_l1_loss(i, t.type_as(i), reduction='none', beta=0)),
+        cpp_function_call='''F::smooth_l1_loss(
+            i, t.to(i.options()), F::SmoothL1LossFuncOptions().reduction(torch::kNone), 0)''',
+        input_fn=lambda: torch.randn(2, 3, 4),
```

Comment:
this won't trigger zero loss, right? If we want to test zero loss, will using `t.clone()` as input work?

Reply:
Two things:

1) I started trying to modify the backwards kernels to avoid/check for divide-by-zeros, but I realized that setting beta=0 is really just the same thing as using l1_loss. I updated the definitions in `Loss.cpp` to check for that case and just call out to their corresponding `l1_loss` functions directly.

2) I added a test to the suite that uses beta=0, but I'm not testing for the case that the two tensors are the same. That's because it doesn't look like that test will play well with this test suite. The test suite asserts the logic by computing the analytical jacobian of the loss (using my backwards) as well as the numerical jacobian (using (f(x+e)-f(x-e)/(2*e))), but that logic breaks down when one of the functions isn't continuous. In this case, the second derivative of smooth_l1_loss when beta=0 isn't continuous- it's equal to infinity at x=0, and 0 everywhere else.